rust/tests/chunks.rs

```diff
@@ -0,0 +1,50 @@
+mod common;
+
+use common::*;
+
+use std::io::BufWriter;
+
+use anyhow::Result;
+use itertools::Itertools;
+use memmap::Mmap;
+use tempfile::tempfile;
+
+/// Checks that the writer will automatically close chunks when they hit a target size.
+#[test]
+fn auto_cut_chunks() -> Result<()> {
+    let mapped = mcap_test_file()?;
+
+    let messages = mcap::MessageStream::new(&mapped)?;
+
+    let mut tmp = tempfile()?;
```

Comment:
seems like this could be a bit cleaner (and less `unsafe`) if we just write to memory rather than a temp file

Reply:
Yeah, that's true of all the tests. I think we were originally testing with larger inputs, and I didn't want to assume they could fit in-memory. But now that `mcap_test_file()` is 2 meg, we should just write to a `Vec<u8>`.

I can open a PR to do that unless one of you wants to beat me to it.

Reply:
Yeah, that's true of all the tests. I think we were originally testing with larger inputs, and I didn't want to assume they could fit in-memory. But now that `mcap_test_file()` is 2 meg, we should just write to a `Vec<u8>`.

I can open a PR to do that unless one of you wants to beat me to it.

Reply:
I'll do that here.